### DISCUSSION TITLE
Wave.js version 2.0.0 feature requests

### DISCUSSION THREAD

**Discussion by foobar404** [foobar404/Wave.js#44] [Ideas]
Hey guys, iv been away for a while, but the library is still receiving decent interest so I have decided to return and create an updated version of the library. This version wont be backwards compatible, and is largely a rewrite. I'm interested in hearing what features if any people would like in this updated version. Please leave a comment below with any ideas you have. 

Below are some features I currently plan on implementing.

* rewrite library in typescript
* use webpack instead of rollup
* make animations composable, so you can join multiple simple animation into something complex
* add support for gradients 
* add support for glow effects
* add options for all wave animations, such as position, rotation, scale, point count, etc..

**Comment by jailbrokebloke** [comment]
Woot, this is exciting — and all those features sound great! I absolutely love the plugin, and really appreciate your work and efforts.

**Comment by NElnour** [comment]
Is it possible to add support for vertical coloring as well for 2.0.0?

Edit: also support for changing bar widths.

**Reply by foobar404** [reply]
both of these features are supported in v2.0, its taking me longer to release this next version then I would like though.

**Comment by mo9a7i** [comment]
I would be very much interested in posibility to have it run back-end as well, to develop pictures of each frame or videos. Using a headless browser probably ?

**Reply by foobar404** [reply]
I was asked about this once, I think it might be possible if you use the canvas npm package.

**Reply by valerii15298** [reply]
Do not know if I understand you correctly, but if you meant for it to work on desktop, there is [electron.js](https://www.electronjs.org/).
And integrating with it is very simple.

**Comment by valerii15298** [comment]
@foobar404 
In the previous version I used this config:
```json
{
  "type": "dualbars blocks",
  "colors": ["green", "lightgreen"],
  "stroke": 3
}
```

Is it possible to use the same animation in new version of wave-js?
Thanks in advance:)

**Reply by foobar404** [reply]
```javascript
new wave.animations.Lines({
    lineWidth: 3,
    lineColor: { gradient: ["green", "lightgreen"] },
    center: true,
    mirroredY: true
})
```

this might get you close

**Comment by liviakuhn** [comment]
Will it be possible to create a waveform visualisation from an audio **stream** also ?
That's what I needed so I made a fork that takes an audio stream instead of an audio element: [liviakuhn/Wave.js](https://github.com/liviakuhn/Wave.js)
Thanks in advance!

**Reply by Tophernl** [reply]
Same here. Could you maybe make a PR for that functionality @liviakuhn ?
